### PR TITLE
Add TS support to ApiView

### PR DESCRIPTION
**Pull request by pakrym**
To upload an API review enable the docModel output in api-extractor (https://api-extractor.com/pages/setup/generating_docs/) and upload resulting `*.api.json` file.

``` json
  "docModel": {
    "enabled": true
  },
```

image

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Languages/JavaScriptLanguageService.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace APIViewWeb
+{
+    public class JavaScriptLanguageService : LanguageProcessor
+    {
+        public override string Name { get; } = "JavaScript";
+        public override string Extension { get; } = ".api.json";
+        public override string ProcessName { get; } = "node";
+        public override string VersionString { get; } = "1";
+
+        public override string GetProccessorArguments(string originalName, string tempDirectory, string jsonFilePath)
```
Comment (JoshLove-msft) [inline src/dotnet/APIView/APIViewWeb/Languages/JavaScriptLanguageService.cs:16]: nit: typo

Comment (pakrym) [reply src/dotnet/APIView/APIViewWeb/Languages/JavaScriptLanguageService.cs:16]: where?

Comment (JoshLove-msft) [reply src/dotnet/APIView/APIViewWeb/Languages/JavaScriptLanguageService.cs:16]: GetProccessorArguments => GetProcessorArguments

File: src/dotnet/APIView/APIViewWeb/APIViewWeb.csproj [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <PublishRepositoryUrl>true</PublishRepositoryUrl>
     <JavaProcessor>..\..\..\java\apiview-java-processor\target\apiview-java-processor-1.9.0.jar</JavaProcessor>
     <PythonProcessor>..\..\..\..\packages\python-packages\api-stub-generator\dist\api_stub_generator-0.1.0-py3-none-any.whl</PythonProcessor>
+    <JaveScriptProcessor>..\..\..\ts\ts-genapi\</JaveScriptProcessor>
```
Comment (AlexGhiondea) [inline src/dotnet/APIView/APIViewWeb/APIViewWeb.csproj:11]: JavaScript?

File: src/ts/ts-genapi/export.ts [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        case ApiItemKind.Interface:
+        case ApiItemKind.Class:
+            typeKind = item.kind.toLowerCase();
+            break
+        case ApiItemKind.TypeAlias:
+            typeKind = "struct";
+            break
+    }
+
+    if (typeKind != null)
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:284]:
```suggestion
    if (typeKind)
```

File: src/ts/ts-genapi/export.ts [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+            }
+
+            var tokens: any[] = Array.from(jsTokens(line));
+            tokens.forEach(token => 
+            {
+                if (this.keywords.indexOf(token.value) > 0)
+                {
+                    this.keyword(token.value);
+                }
+                else if (token.value == currentTypeName)
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:189]:
```suggestion
                else if (token.value === currentTypeName)
```

Basically never use `==` in JS, always use `===`

File: src/ts/ts-genapi/export.ts [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+        [propertyName: string]: string;
+    };
+}
+
+function appendMembers(builder: TokensBuilder, navigation: IApiViewNavItem[], item: ApiItem)
+{
+    builder.lineId(item.canonicalReference.toString());
+    builder.indent();
+    if (item instanceof ApiDeclaredItem) {
+        item.excerptTokens.forEach(token => {
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:258]:
```suggestion
       for(const token of item.excerptTokens) {
```

Comment (xirzec) [reply src/ts/ts-genapi/export.ts:258]: Prefer for/of to .forEach

File: src/ts/ts-genapi/export.ts [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+                .newline();
+        }
+    }
+    else
+    { 
+        builder.newline();
+    }
+}
+
+const apiModel: ApiModel = new ApiModel();
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:332]:
```suggestion
const apiModel = new ApiModel();
```

no need to annotate types when they're directly inferable

File: src/ts/ts-genapi/export.ts [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+    ApiItemKind,
+    ApiDeclaredItem,
+    ExcerptTokenKind
+  } from '@microsoft/api-extractor-model';
+
+import {writeFile} from 'fs';
+
+const jsTokens = require("js-tokens");
+
+class TokensBuilder
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:13]: there's a lot going on in this file -- is it possible to break it into modules and put `TokensBuilder` in its own file at least?

File: src/ts/ts-genapi/export.ts [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+        )
+);
+
+var apiViewFile: IApiViewFile = {
+    Name: apiModel.packages[0].name,
+    Navigation: navigation,
+    Tokens: builder.tokens
+}
+
+writeFile(process.argv[3], JSON.stringify(apiViewFile), err => {
```
Comment (xirzec) [inline src/ts/ts-genapi/export.ts:351]: if you want to use async/await instead of callbacks: https://masteringjs.io/tutorials/node/promisify

**Pull request merged**